# HG changeset patch
# User Rik <code@nomad.inbox5.com>
# Date 1271991675 25200
#      Thu Apr 22 20:01:15 2010 -0700
# Node ID 83de7b060e91f61801fe93b86623ff7d283d14f0
# Parent  e9d1a974a517f2bb7ac02764d7f6174c94e210b6
Return correct value (101) from sqp on succes.  Fixes bug #29577.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2010-04-22  Rik <octave@nomad.inbox5.com>
+
+	* optimization/sqp.m: Return correct value (101) on succes.  Fixes bug #29577
+
 2010-04-22  Rik <octave@nomad.inbox5.com>
 
 	* optimization/glpk.m: Eliminate repmat (Inf, x , y) and just use Inf (x,y)
 
 2010-04-22  Rik <octave@nomad.inbox5.com>
 
 	* general/rat.m, sparse/gplot.m, strings/base2dec.m, geometry/triplot.m
      optimization/sqp.m, optimization/qp.m, statistics/base/run_count.m,
diff --git a/scripts/optimization/sqp.m b/scripts/optimization/sqp.m
--- a/scripts/optimization/sqp.m
+++ b/scripts/optimization/sqp.m
@@ -392,16 +392,17 @@ function [x, obj, info, iter, nf, lambda
 
       t0 = norm (c - A' * lambda);
       t1 = norm (ce);
       t2 = all (ci >= 0);
       t3 = all (lambda_i >= 0);
       t4 = norm (lambda .* con);
 
       if (t2 && t3 && max ([t0; t1; t4]) < tol)
+        info = 101;
 	break;
       endif
 
       ## Compute search direction p by solving QP.
 
       g = -ce;
       d = -ci;
 
